server/datastore/mysql/software_title_icons.go

```diff
@@ -52,6 +52,63 @@ func (ds *Datastore) GetSoftwareTitleIcon(ctx context.Context, teamID uint, titl
 	return &icon, nil
 }
 
+func (ds *Datastore) GetSoftwareIconsByTeamAndTitleIds(ctx context.Context, teamID uint, titleIDs []uint) ([]fleet.SoftwareTitleIcon, error) {
+	var args []interface{}
+	query := `
+		SELECT team_id, software_title_id, storage_id, filename
+		FROM software_title_icons
+		WHERE software_title_id IN (?) AND team_id = ?
+	`
+	query, args, err := sqlx.In(query, titleIDs, teamID)
+	if err != nil {
+		return nil, ctxerr.Wrap(ctx, err, "building query for get software title icons")
+	}
+
+	var icons []fleet.SoftwareTitleIcon
+	if err := sqlx.SelectContext(ctx, ds.reader(ctx), &icons, query, args...); err != nil {
+		return nil, ctxerr.Wrap(ctx, err, "get software title icons")
+	}
+
+	return icons, nil
+}
+
+func (ds *Datastore) GetSoftwareTitleIconsByTeamAndAdamIDs(ctx context.Context, teamID uint, adamIDs []string) (map[string]fleet.SoftwareTitleIcon, error) {
```

Comment:
Looking at this more closely, thinking that if we needed this we would want to call this based on title ID (so, unifying calls by title ID) rather than having a separate path here. This is because filtering by team ID and a list of adam IDs will give us multiple icons in cases where iOS/iPadOS/macOS icons are set and different. So this signature basically forces the implementation to be bugged, and we don't actually use anything from the VPP apps table other than the adam ID index itself.

But also, if we don't need to override icons in the VPP apps list endpoint (IMO we don't, and it's not in the spec), we don't need this function at all anyway.